### PR TITLE
add answerYaml override support

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
need https://github.com/rancher/helm/pull/7 and https://github.com/rancher/types/pull/568.
This pr adds the following features:
1. add a field called `answerYaml` which allows user to post a values.yml file to override default one.
2. Add a custom parser to identify if answers has prefix like (string:test1234), and convert those values accordingly. If not, fall back to the old one.

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/13158

File: pkg/controllers/user/helm/common/common.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ import (
 	"strings"
 	"time"
 
+	"io/ioutil"
```
Comment (dramich) [inline pkg/controllers/user/helm/common/common.go]: Fix formatting

File: pkg/controllers/user/helm/common/common.go [hunk 2]
```diff
@@ -107,3 +105,23 @@ func DeleteCharts(port string, obj *v3.App) error {
 	}
 	return errors.New(string(combinedOutput))
 }
+
+func GenerateAnswerSetValues(app *v3.App, tempDir string) ([]string, error) {
+	setValues := []string{}
+	if app.Spec.AnswersYaml != "" {
```
Comment (dramich) [inline pkg/controllers/user/helm/common/common.go]: naming?

File: tests/core/test_default_roles.py [hunk 3]
```diff
@@ -103,7 +103,7 @@ def test_project_create_default_role(admin_mc, cleanup_roles, remove_resource):
 
     set_role_state(client, test_roles, 'project')
 
-    project = client.create_project(name=random_str(), clusterId='local')
+    project = client.create_project(name=random_str(), clusterId=CLUSTER_ID)
```
Comment (dramich) [inline tests/core/test_default_roles.py]: Use the clusterID from the client

File: tests/core/test_default_roles.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import pytest
 import json
 from .common import random_str
-from .conftest import wait_for_condition, wait_until
+from .conftest import wait_for_condition, wait_until, CLUSTER_ID
```
Comment (dramich) [inline tests/core/test_default_roles.py]: Don't import CLUSTER_ID
(ditto for other files)

**Comment by alena1108** [comment]
Closing in favor of https://github.com/rancher/rancher/pull/17576

**Pull request closed without merging**